src/foam/core/reflow/ReflowDashboardView.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.reflow',
+  name: 'ReflowDashboardView',
+  extends: 'foam.dashboard.view.Dashboard',
+  // extends: 'foam.u2.View',
+  mixins: ['foam.u2.layout.ContainerWidth'],
+
+  imports: [
+    'displayWidth?',
+    'document'
+  ],
+
+  css: `
+    ^ {
+      height: 100%;
+      width: 100%;
+      display: flex;
+      flex-direction: column;
+    }
+    ^main {
+      height: fit-content;
+      min-height: 600px;
+      padding: 24px 32px;
+      max-width: 160rem;
+      margin: auto;
+    }
+    ^widget-container {
+      width: 100%;
+      display: grid;
+      flex-grow: 1;
+    }
+  `,
+
+  listeners: [
+    {
+      name: 'updateCols',
+      on: 'data.propertyChange.widgets',
+      isFramed: true,
+      code: function() {
+        let cw = this.containerWidth;
+        let currentWidgetSet = 0;
+        let widgetSetCount = 0;
+        let cm = {};
+        
+        if ( this.data && this.data.widgets ) {
+          this.data.widgets.forEach(widget => {
+            let col = widget[`${cw}Column`] ?? widget['column'];
+            if ( foam.String.isInstance(col) ) {
+              let colSpan = col.indexOf('fr') != -1 ? col.split('fr')[0] : 1;
+              if ( widgetSetCount == 0 )
+                currentWidgetSet++;
+              widgetSetCount += Number(colSpan);
+              cm[widget.id] = `span calc(${colSpan}*var(--dashboard-max-col)/var(--split-row-${currentWidgetSet}))`;
+            } else {
+              if ( widgetSetCount > 0 )
+                this.document.documentElement.style.setProperty(`--split-row-${currentWidgetSet}`, widgetSetCount); 
+              widgetSetCount = 0;
+              cm[widget.id] = 'span ' + col;
+            }
+          });
+        }
+        
+        this.containerMap = cm;
+      }
+    }
+  ],
+
+  properties: [
+    {name:'data'},
```

Comment:
shorter form is just: "'data'", if you aren't setting any properties other than name:.